<div id="topHeader"></div>

<!-- https://jquerytools.github.io/documentation/rangeinput/index.html might help us improve this (so it looks the same on all platforms)-->
<div id="thresholdValue">{{threshold | number:2}}</div>
<input id="thresholdRange" ng-model="threshold" type="range" step="0.01" min="0.03" max="{{maxRangeBound}}">
<div id="sliderMessageBox" align="right" ng-show="lowEdgeCount">
  <div class="arrow-up" style="margin-right: 10px"></div>
  <div id="sliderMessageText">Tip: lower the threshold to see more edges...</div>
</div>
<div id="rightRangeLabel">sparse</div>
<div id="leftRangeLabel">dense</div>

<div id="settingsBacking" ng-show="settingsOpen || helpOpen" ng-click="settingsOpen = false; helpOpen = false"></div>
<div id="settingsPanel" ng-show="settingsOpen">
  cluster results by
  <select ng-model="groupedType">
    <option value="cellType">Cell Type</option>
    <option value="lab">Lab</option>
    <option value="organism">Organism</option>
    <option value="treatments">Treatments</option>
    <option value="lifeStage">Life Stage</option>
    <option value="cellTypeAndTreatments">Cell Type and Treatments</option>
  </select><br/>
  <br/>
  <!-- <input type="checkbox" ng-model="showCrossLinks" /> show links between clusters<br/>
  <br/> -->
  <input type="checkbox" ng-model="includeNearby" /> include nodes connected to search matches<br/>
  <br>
  <input type="checkbox" ng-model="markBioGRIDEdges" /> mark edges representing known interactions in BioGRID<br/>
  <br>
  show no more than <input type="text" size="4" ng-model="maxMatches" /> nodes to allow for smooth animation<br/>
  <br>
  <div align="center">
    <button type="button" class="btn btn-default btn-sm" ng-click="downloadGraph()">
      <span class="glyphicon glyphicon-cloud-download" aria-hidden="true"></span> Download current graph view
    </button>
  </div>
</div>
<div id="helpPanel" ng-show="helpOpen">
  <div class="panel panel-default" style="height: 100%;">
    <div class="panel-body" style="height: 100%; overflow-y: scroll">
      <center><h3 style="color: #666">Interface documentation</h3></center>
      <br>
      <h4 style="font-weight: bold">Search</h4>
      <p>Every dataset in ChromNet has associated metadata that is used to return search results in the interface. To search all datasets simply start typing in the search field. Datasets from <a href="http://encodeproject.org" target="_top">ENCODE</a> include a title, description, lab name, cell type, accession id, etc. while custom datasets include metadata provided during network generation.</p>

      <p>Matching search results appear as standard sized nodes in the network. When the "include nearby nodes" option is enabled in settings (it is by default) then datasets that do not match the search but are connected to a search match through an edge are shown as smaller nodes in the network. These "nearby" nodes appear and disappear when you adjust the edge threshold slider.</p>
      <br>
      <h4 style="font-weight: bold">Edge threshold</h4>
      <p>Every edge in the ChromNet group graphical model has an associated score. High magnitude scores represent stronger statistical dependencies, and are also more likely to represent biologically meaningful interactions. Sliding the edge threshold control in the top right of the interface controls which network edges are shown in the network. Dragging to the far right shows only the strongest edges, while dragging to the far left shows even the most weak speculative edges. The lower bound on the edge threshold was set at 0.03 because below this threshold there was no enrichment for known protein-protein interactions among edges inferred using ENCODE data tracks.</p>

      <p>The exact score of an edge can be determined by hovering the mouse over the edge. This causes a short description, including the edge's score, to be displayed in the lower left of the screen.</p>
      <br>
      <h4 style="font-weight: bold">Color clusters</h4>
      <p>To help organize the network during visualization datasets are classified into a set of colored clusters, where only edges within a cluster are shown. To change how this clustering is performed, open the settings gear at the top right of the interface and choose a different clustering criteria.</p>

      <p>The color cluster groups currently shown are listed on the right of the interface. Hovering over a node underlines the corresponding color cluster name, and clicking on any of the color cluster names adds it to the search term, restricting the results to only that color cluster.</p>
      <br>
      <h4 style="font-weight: bold">Dataset inspection</h4>
      <p>Clicking on a dataset in the interface selects that node and opens a detail panel at the bottom of the screen. This panel displays metadata and links to the original data source. When a node is selected all other nodes with the same label (typically the target protein name) are also shown in currently visible color clusters. This allows easy inspection of whether a certain protein target is not displayed because it is not "nearby" (connected by an edge to a search match) or because no such experiment exists. To de-select a dataset simply click anywhere in the background.</p>

      <p>Note that nodes in the interface can be dragged and zooming is supported through the mouse wheel.</p>
      <br>
      <h4 style="font-weight: bold">Graph download</h4>
      <p>To download a spreadsheet representing the currently displayed sub-network, open the settings panel at the top right of the screen and click "Download current graph view." This will save a CSV file to your local computer with the datasets groups and edges that match your current search.</p>
      <br>
      <h4 style="font-weight: bold">Custom datasets</h4>
      <p>To incorporate your own custom datasets first download and install the <a href="https://github.com/slundberg/ChromNet.jl" target="_top">ChromNet processing software</a>, then after generating a custom network JSON file, drag this file to the box titled "Drop custom network JSON files here for visualization" on the ChromNet interface. Note this box is only visible when no search query is present.</p>
    </div>
  </div>
</div>
<img id="settingsIcon" src="/image_files/cog_small.png" ng-click="settingsOpen = !settingsOpen" />
<img id="helpIcon" src="/image_files/helpIcon.png" ng-click="helpOpen = !helpOpen" />

<div id="searchBoxWrapper"><div ng-show="searchString != '' && numMatches <= maxMatches" style="position: absolute; margin-left: -50px; right: 0px; overflow: visible; padding: 4px; text-align: right; padding-right: 10px; padding-top: 5px; color: #aaa; font-size: 14px;">{{numHighlightMatches}} matches, {{numNearbyMatches}} nearby</div><input type="text" id="searchBox" style="" ng-model="searchString" autofocus placeholder="type to search..."/></div>
<img id="linkLogo" src="/image_files/chromNetLogo.png" ng-click="searchString = ''" />

<feature-network nodes="presentedNodes" links="presentedLinks" type-map="typeMap" selected-node="selectedTrack" hover-node="hoverNode" hover-edge="hoverEdge"></feature-network>

<div id="blankScreenMessage" ng-hide="presentedNodes.length || searchString">
  <span style="color: #333">Type to search {{leafCount}} {{customData ? 'data' : 'ENCODE ChIP-seq'}} tracks, or try an example query such as <a href ng-click="searchString = 'EZH2'">EZH2</a>, <a href ng-click="searchString = 'H3K4me3 K562'">H3K4me3 K562</a>, or <a href ng-click="searchString = 'MYC NB4'">MYC NB4</a>.</span><br>
  <br>
  <div style="font-size: 13px; text-align: justify;">ChromNet is designed to identify direct interactions between ChIP-seq datasets, which are correlations unique to a pair of variables that cannot be explained by any other dataset in the database. ChromNet uses a group graphical model representation that integrates both a hierarchical clustering of the data represented by color intensity, and direct interactions between groups represented by edges. For more details and raw data see the <a href="https://sites.google.com/a/cs.washington.edu/chromnet/" target="_top">paper to come</a> and the online <a href=""  ng-click="helpOpen = !helpOpen">documentation</a>. Source code for both the <a href="https://github.com/slundberg/ChromNet.js" target="_top">client interface</a> and to <a href="https://github.com/slundberg/ChromNet.jl" target="_top">generate custom ChromNet networks</a> is available on GitHub.</div><br>
  <br>
  <br>

  <div id="customNetworkDropZone" style="color: {{uploadColor}}; border-color: {{uploadColor}};">
    {{uploadStatusMessage}}
    <!--(your network data is only used locally and never sent to our servers.)-->
  </div>
  <div id="list"></div>
  <a href="http://www.cs.washington.edu"><img src="image_files/cseLogo.png" style="margin-top: 50px;"/></a>
</div>

<div id="blankScreenMessage" style="color: #aaa" ng-show="numMatches > maxMatches">
  Keep typing to refine your search, or raise the edge threshold. Currently over {{maxMatches}} tracks match your query...
</div>

<div id="cellTypeList">
  <span ng-hide="types.length == 0" style="color: #999; line-height: 150%; font-weight: normal;">{{groupedTypeNames[groupedType]}}</span><br>
  <span ng-repeat="type in types"><nobr><span ng-click="addSearchTerm(type.name)" style="pointer-events: all; cursor: pointer; color: {{type.color.toString()}}" class="underline-{{hoverNode.type == type.name}}">{{type.name}}</span></nobr><br></span>
</div>


<div style="position: absolute; left: 0px; right: 0px; bottom: 0px;">
  <div class="rolloverInfo" ng-show="hoverNode" style="opacity: 0.7; color: {{typeMap[hoverNode.type].color}};">
  {{hoverNode.data.description ? hoverNode.data.description : "ChIP-seq on " + hoverNode.data.cellType }} <span style="font-style: italic; font-weight: normal">- {{hoverNode.data.lab}}</span>
  </div>
  <div class="rolloverInfo" ng-show="hoverEdge" style="font-weight: normal; opacity: 0.7; color: #666;">
  {{hoverEdge.source.name}} -- {{hoverEdge.target.name}} edge weight is {{hoverEdge.coeff | number:3}} <span ng-repeat="label in hoverEdge.labels">({{label}})</span>
  </div>
  <!-- <div style="background: linear-gradient(rgba(239, 239, 239, 0.0),rgb(239,239,239); rgba(239, 239, 239, 1)); height: 15px;"></div> -->
  <div ng-show="selectedTrack && selectedTrackColor" style="background: rgba({{selectedTrackColor.r}},{{selectedTrackColor.g}},{{selectedTrackColor.b}},0.9); padding: 15px; padding-top: 0px; user-select: text">
    <div style="height: 25px; color: #fff; font-weight: bold; text-align: center; line-height: 25px">{{selectedTrack.data.description}} <span style="font-weight: normal; font-style: italic">- {{selectedTrack.data.lab}}</span></div>
    <table style="width: 100%"><tr>
      <td style="width: 180px; vertical-align: top">
        <b style="color: #fff">Experiment</b><br>
        <a class="grayLink" href="https://www.encodedcc.org/experiments/{{selectedTrack.data.id}}" target="_new">{{selectedTrack.data.id}}</a><br>
        <div style="height: 5px;"></div>
        <b style="color: #fff">Antibody</b><br>
        <a class="grayLink" href="https://www.encodedcc.org/search/?searchTerm={{selectedTrack.data.antibody}}&type=antibody_approval" target="_new">{{selectedTrack.data.antibody}}</a>

      </td>
      <td style="vertical-align: top; color: #eee;">
        <img src="image_files/close.png" ng-click="deselectTrack()" style="float: right; opacity: 0.2; margin-right: -4px; margin-top: -14px; cursor: pointer" />
        <b style="color: #fff">{{selectedTrack.annotationSymbol}}</b> -
        {{selectedTrack.annotationSummary}}
      </td>
    </tr></table>
  </div>
</div>


<!-- <div style="position: absolute; left: 0px; top: {{windowHeight()-25}}px; width: 100%; height: 25px; overflow: hidden;" id="ucscWrapper">
<div style="width: 100%; height: 25px; background-color: #1B71F1; text-align: center;" id="ucscResize" class="ui-resizable-handle ui-resizable-n opacity-{{ucscClosed}}">
  <span style="line-height: 25px; color: #fff; opacity: 0.9">UCSC Genome Browser (hg19)</span>
</div>
<div style="z-index: 2; position: relative; top: 0px; left: 0px; width: 100%; height: 5px; background: url(http://genome.ucsc.edu/image_files/internal_bg1.jpg) repeat-x fixed left bottom"></div>
<div style="position: relative; left: 0px; bottom: 0px; width: 100%; height: 25px; overflow: hidden;" class="ucscResizeAlso">
<iframe border="0" class="ucscResizeAlso" style="overflow-x: hidden; overflow-y: scroll; border: none; position: relative; left: 0px; top: -64px; height: 60px; width: 100%" ng-src="{{ucscUrl}}"></iframe>
</div>
</div> -->




<!-- <div style="position: absolute; display: block; bottom: 305px; left: 0px; width: 100%; height: 25px; background-color: rgb(219, 223, 226)"></div>
<div style="position: absolute; display: block; bottom: 312px; left: 0px; width: 100%; height: 30px; background-color: #1B71F1" ></div>

 -->
<!--<div id="trackInspector" class="s">
  <div style="z-index: 100; height: 20px; margin-bottom: -21px; text-align: center; margin-left: -120px; width: 240px; font-size: 12px; color: #999; font-weight: bold; position: relative; top: 0px; left: 50%"><span style="background: linear-gradient(to right, rgba(255, 255, 255, 0.2), rgba(255, 255, 255, 1), rgba(255, 255, 255, 1), rgba(255, 255, 255, 1), rgba(255, 255, 255, 1), rgba(255, 255, 255, 0.2)); padding-left: 70px; padding-right: 70px;">Chromosome 10</span></div>

  <div ng-repeat="track in selectedTracks" class="slide-repeated" style="margin-top: 0px;">

    <scalable-series style="width: 100%; height: 80px; margin-bottom: -20px;" view="sharedZoom" hide-x-axis="{{!$first}}" height="{{$first ? 80 : 60}}" track="minmax.{{track.data.id}}.chr10" color="{{typeMap[track.data.cellType].color}}"></scalable-series>

    <div style="position: relative; z-index: 100; pointer-events: none; margin: auto; margin-bottom: 0px; white-space: nowrap; text-align: center; color: #999; font-size: 12px"><span ng-click="track.showGeneAnnotation = !track.showGeneAnnotation" style="pointer-events: all; cursor: pointer; font-weight: bold; color: {{typeMap[track.data.cellType].color}}" >{{track.data.description}}</span>, <span style="font-size: 11px;">{{track.data.lab}}</span></div>

    <img src="image_files/closeSmaller.png" ng-click="closeTrack(track)" style="pointer-events: all; float: right; opacity: 0.25; margin-top: -13px; margin-right: 4px; cursor: pointer" />

    <div ng-show="track.showGeneAnnotation">
      <div style="background: linear-gradient(rgba(239, 239, 239, 0.0), rgba(239, 239, 239, 1)); height: 15px;"></div>
      <div style="background: rgb(239,239,239); padding: 15px; padding-top: 0px; user-select: text">
        <table style="width: 100%"><tr>
          <td style="width: 180px; vertical-align: top">
            <b style="color: #999">Experiment</b><br>
            <a class="grayLink" href="https://www.encodedcc.org/experiments/{{track.data.id}}" target="_new">{{track.data.id}}</a><br>
            <div style="height: 5px;"></div>
            <b style="color: #999">Antibody</b><br>
            <a class="grayLink" href="https://www.encodedcc.org/search/?searchTerm={{track.data.antibody}}&type=antibody_approval" target="_new">{{track.data.antibody}}</a>

          </td>
          <td style="vertical-align: top;">
            <img src="image_files/close.png" ng-click="track.showGeneAnnotation = false" style="float: right; opacity: 0.2; margin-right: -4px; cursor: pointer" />
            <b style="color: #999">{{track.annotationSymbol}}</b><br>
            {{track.annotationSummary}}
          </td>
        </tr></table>
      </div>
    </div>
  </div>
</div>-->
